rastervision/backend/torch_utils/chip_classification/data.py

```diff
@@ -3,19 +3,49 @@
 
 import numpy as np
 import torch
+
 from torch.utils.data import DataLoader, Dataset
+from torch.utils.data.sampler import WeightedRandomSampler
 from albumentations.core.composition import Compose
 from albumentations.augmentations.transforms import (
     Blur, RandomRotate90, HorizontalFlip, VerticalFlip, GaussianBlur,
     GaussNoise, RGBShift, ToGray)
 
 from rastervision.backend.torch_utils.data import DataBunch
-from rastervision.backend.torch_utils.chip_classification.folder import (
-    ImageFolder)
+from rastervision.backend.torch_utils.chip_classification.folder import ImageFolder
 
 log = logging.getLogger(__name__)
 
 
+def calculate_oversampling_weights(imageFolder, rare_classes, desired_prob):
+    '''
+    Calculates weights tensor for oversampling
+
+    args:
+        imageFolder: instance of
+            rastervision.backend.torch_utils.chip_classification.folder.ImageFolder
+        rare classes: (list) of ints of the classes that should be oversamples
+        desired prob: (float) a single probability that the rare classes should
+            have.
+    returns:
+        (tensor) with weights per index, e.g [0.5,0.1,0.9]
+    '''
+
+    chip_inds = []
+    for rare_class_id in rare_classes:
+        for idx, (sample, class_idx) in enumerate(imageFolder):
+            if class_idx == rare_class_id:
```

Comment:
You could remove the outer loop over `rare_classes` if you change this line to `if class_idx in rare_classes:`

Reply:
Done